client/app/components/Topbar/Logo.js

```diff
@@ -0,0 +1,36 @@
+import { Component, PropTypes } from 'react';
+import r, { div, a } from 'r-dom';
+import classNames from 'classnames';
+
+import css from './Logo.css';
+
+const logoContent = function logoContent(image, text) {
+  let content = null;
+  if (image) {
+    content = r.img({ src: image, className: css.logoImage });
+  } else if (text) {
+    content = r.span({ className: css.logoText }, text);
+  }
+  return content;
+};
+
+class Logo extends Component {
+  render() {
+    return a({
+      className: classNames(this.props.className, css.logo),
+      href: this.props.href,
+    }, logoContent(this.props.image, this.props.text));
+  }
+}
+
+Logo.propTypes = {
+  href: PropTypes.string.isRequired,
+  image: PropTypes.string,
+  text: PropTypes.string,
+  className: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.objectOf(PropTypes.bool),
+  ]),
```

Comment:
Maybe not yet since this is the first time this is used, but a named PropType for the className is most likely needed in the future if this becomes a widely used pattern.